flight/PiOS/STM32F4xx/pios_usb_cdc.c

```diff
@@ -463,39 +464,9 @@ static bool PIOS_USB_CDC_CTRL_EP_IN_Callback(uintptr_t usb_cdc_id, uint8_t epnum
 	bool valid = PIOS_USB_CDC_validate(usb_cdc_dev);
 	PIOS_Assert(valid);
 
-	/* Give back UART State Bitmap */
-	/* UART State Bitmap
-	 *   15-7: reserved
-	 *      6:  bOverRun    overrun error
-	 *      5:  bParity     parity error
-	 *      4:  bFraming    framing error
-	 *      3:  bRingSignal RI
-	 *      2:  bBreak      break reception
-	 *      1:  bTxCarrier  DSR
-	 *      0:  bRxCarrier  DCD
-	 */
-
-	/* Currently, we only handle TxCarrier and RxCarrier reporting */
-	uint16_t new_uart_state = 0;
-	if (usb_cdc_dev->tx_out_cb) {
-		/* Someone is going to providing FC->PC data, advertise an RxCarrier to the host */
-		new_uart_state |= 0x1;
-	}
-	if (usb_cdc_dev->rx_in_cb) {
-		/* Someone is consuming PC->FC data, advertise a TxCarrier to the host */
-		new_uart_state |= 0x2;
-	}
-
-	/* Has anything changed since we last sent a notification? */
-	if ((new_uart_state ^ usb_cdc_dev->prev_uart_state) & 0x3) {
-		usb_cdc_dev->prev_uart_state = new_uart_state;
-
-		uart_state.bmUartState = htousbs(new_uart_state);
-
-		PIOS_USBHOOK_EndpointTx(usb_cdc_dev->cfg->ctrl_tx_ep,
-					(uint8_t *)&uart_state,
-					sizeof(uart_state));
-	}
+	PIOS_USBHOOK_EndpointTx(usb_cdc_dev->cfg->ctrl_tx_ep,
+			(uint8_t *)&uart_state,
+			sizeof(uart_state));
```

Comment:
Just to be clear-- repeated control messages with then blhelisuite bailing out and saying it couldn't connect.

I referred to the spec and other ref code (linux kernel).  I believe, but can't be certain, that this is the minimal changeset required for things to consistently work.  Ideally we'd better notice connect/disconnect and ensure to notify at least once in the beginning-- but I'm not even positive this is fully safe.

In any case, -- it doesn't increase the number of transfers (notifications are polled, and declining to notify is a transfer in itself), and client code properly looks for edges (it has to-- if there's another interrupt it'll still receive a duplicate on the other bits).  If we want to lower bus overhead, we could lower bInterval on this mgmt endpoint, which is currently 4ms.